### PR TITLE
[KDA] Support extra_buffer mamba radix cache for Kimi Linear

### PR DESCRIPTION
**Pull request by yyq0210**
## Motivation

KDA (Kimi Delta Attention) previously did **not** support the `extra_buffer` mamba radix cache strategy. Forcing it on regressed GSM8K accuracy to **~0.63**, so `KimiLinearForCausalLM` fell back to `no_buffer`, which requires `--disable-overlap-schedule` and gives up the overlap throughput win.

  **Root cause:** `extra_buffer` prefix caching needs the intermediate SSM state `h` at chunk boundaries to be tracked so radix can restore a shared prefix. This was wired for GDN / Mamba2 but not for KDA, so the tracked prefix slots held stale/zero state and every prefix-reusing request restored a garbage SSM state.

  **Key point:** KDA's `chunk_kda_fwd` *already computes* the same per-chunk-boundary `h` as GDN's chunk kernel (`chunk_gated_delta_rule_fwd_h`) — it was simply discarded. So this fix is **pure plumbing: no new or modified Triton kernel.**

## Modifications

 - **`fla/kda.py`** — `chunk_kda_fwd` / `chunk_kda` gain `output_intermediate_states`; return `(o, h)` when set, else `o` unchanged.
  - **`kernels/kda_triton.py`** — `TritonKDAKernel.extend` threads the flag and returns `(core_attn_out, h)`.
  - **`kernels/kda_cutedsl.py` / `kernels/kda_flashkda.py`** — assert-guard the flag, since the fused kernels do not expose `h` (`extra_buffer` forces the Triton prefill path anyway).
  - **`kda_backend.py`** —
    - store the **transposed** `conv_states_shape` (KDA's raw `conv[0]` is `(slots, K-1, channel)` vs GDN's `(slots, channel, K-1)`), so the shared `_init_track_conv_indices` reads `[-1] == K-1` as the window length;
    - set the track / mask indices in an `init_forward_metadata` override (mirrors GDN);
    - in `forward_extend`, snapshot the pre-conv window into the transposed `conv[0].transpose(-1, -2)` view and track the intermediate SSM state via `_track_mamba_state_extend`.
  - **`arg_groups/overrides.py`** — whitelist `KimiLinearForCausalLM` for `extra_buffer` (gated on `linear_attn_backend == "triton"`).
  - **`test/registered/unit/layers/test_kda_extra_buffer_conv_track.py`** — new CPU unit test asserting the KDA transposed-view conv snapshot is bit-identical to a native GDN `(slots, channel, K-1)` snapshot (track + restore parity).

## Accuracy Tests

**CPU unit test:** 4/4 pass.

  **E2E** on `Kimi-Linear-48B-A3B-Instruct` (TP=2, GSM8K 5-shot, 1319 questions, radix + CUDA graph on, no speculative decoding):

  | Mode | Overlap schedule | Accuracy | Latency |
  |------|------------------|----------|---------|
  | `no_buffer` (baseline) | forced **off** | 0.892 | 97.5 s |
  | `extra_buffer` (this PR) | **on** | **0.900** | **87.4 s** |

  `extra_buffer` now **matches** `no_buffer` accuracy (the 0.63 regression is gone) and runs **~10% faster** by keeping the overlap schedule enabled.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #29228887165](https://github.com/sgl-project/sglang/actions/runs/29228887165)
Latest PR Test (Extra): :x: [Run #29228887089](https://github.com/sgl-project/sglang/actions/runs/29228887089)

### CODE REVIEW

File: python/sglang/srt/layers/attention/linear/kernels/kda_flashkda.py [hunk 1]
```diff
@@ -113,6 +113,11 @@ def extend(
         is_spec_decode: bool = False,
         **kwargs,
     ) -> torch.Tensor:
+        assert not kwargs.get("output_intermediate_states"), (
```
Comment (yuan-luo) [inline python/sglang/srt/layers/attention/linear/kernels/kda_flashkda.py:119]: This guard is too late.
When user chooses --linear-attn-prefill-backend flashkda/cutedsl, and extra_buffer is chosen, we need to report a configure error.

File: test/registered/unit/layers/test_kda_extra_buffer_conv_track.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    def _make_inputs(self):
+        # mixed_qkv_t is `[channel, seq]` (post transpose(0, 1) in forward_extend).
+        mixed_qkv_t = torch.randn(self.channel, self.seq)
+        # Two tracked rows, each a length-(K-1) window of flat positions.
+        starts = torch.tensor([10, 25])
+        track_conv_indices = starts.unsqueeze(-1) + torch.arange(self.k_minus_1)
+        mask_indices = torch.tensor([2, 5])  # physical track-dest slots
+        return mixed_qkv_t, track_conv_indices, mask_indices
+
+    def test_transposed_view_matches_gdn_layout(self):
```
Comment (yuan-luo) [inline test/registered/unit/layers/test_kda_extra_buffer_conv_track.py]: Could you add a test to cover tracking/restoring the intermediate SSM state h?
The PR’s core algorithm depends on the h shape and source indexing matching GDN’s convention. We'd better protect the real failure mode, rather than only covering the conv window transpose.

File: python/sglang/srt/layers/attention/linear/kernels/kda_flashkda.py (hunk 1 above)
Comment (yyq0210) [reply python/sglang/srt/layers/attention/linear/kernels/kda_flashkda.py:119]: Fixed. The gate now checks the effective prefill backend  (`linear_attn_prefill_backend or linear_attn_backend`), and `_validate_mamba_extra_buffer` raises a config-time error naming the offending
backend — so it fails at startup, not in the kernel. The kernel asserts remain only as a backstop.

File: test/registered/unit/layers/test_kda_extra_buffer_conv_track.py (hunk 2 above)
Comment (yyq0210) [reply test/registered/unit/layers/test_kda_extra_buffer_conv_track.py]: Added `TestKDAExtraBufferSSMTrack`: verifies the `h` layout (`[1, NT, HV, V, K]` → `squeeze(0)` → `ssm_states[dst]`) and the KDA ceil-branch source indexing, including a case that fails if KDA is mis-routed onto the Mamba2 floor branch. Merged into one `test_kda_extra_buffer_track.py`.
